jetstream/stream.go

```diff
@@ -259,28 +260,52 @@ func (s *stream) Info(ctx context.Context, opts ...StreamInfoOpt) (*StreamInfo,
 	}
 	var req []byte
 	var err error
-	if infoReq != nil {
-		req, err = json.Marshal(infoReq)
-		if err != nil {
-			return nil, err
-		}
-	}
+	var subjectMap map[string]uint64
+	var offset int
 
 	infoSubject := apiSubj(s.jetStream.apiPrefix, fmt.Sprintf(apiStreamInfoT, s.name))
-	var resp streamInfoResponse
-
-	if _, err = s.jetStream.apiRequestJSON(ctx, infoSubject, &resp, req); err != nil {
-		return nil, err
-	}
-	if resp.Error != nil {
-		if resp.Error.ErrorCode == JSErrCodeConsumerNotFound {
-			return nil, ErrStreamNotFound
+	for {
+		if infoReq != nil {
+			if infoReq.SubjectFilter != "" {
+				if subjectMap == nil {
+					subjectMap = make(map[string]uint64)
+				}
+				infoReq.Offset = offset
+			}
+			req, err = json.Marshal(infoReq)
+			if err != nil {
+				return nil, err
+			}
+		}
+		var resp streamInfoResponse
+		if _, err = s.jetStream.apiRequestJSON(ctx, infoSubject, &resp, req); err != nil {
+			return nil, err
+		}
+		if resp.Error != nil {
+			if resp.Error.ErrorCode == JSErrCodeStreamNotFound {
+				return nil, ErrStreamNotFound
+			}
+			return nil, resp.Error
+		}
+		s.info = resp.StreamInfo
+		var total int
+		if resp.Total != 0 {
+			total = resp.Total
+		}
+		if len(resp.StreamInfo.State.Subjects) > 0 {
+			for subj, msgs := range resp.StreamInfo.State.Subjects {
+				subjectMap[subj] = msgs
+			}
+			offset = len(subjectMap)
+		}
+		if total == 0 || total <= offset {
+			resp.StreamInfo.State.Subjects = subjectMap
+			s.info = resp.StreamInfo
+			break
 		}
-		return nil, resp.Error
 	}
-	s.info = resp.StreamInfo
 
-	return resp.StreamInfo, nil
+	return s.info, nil
```

Comment:
Should we consider iterator API for receiving those subjects to avoid allocating a big map?

Reply:
That would change the API - and require someone to page it manually - the big map is going to be large, but that in the end is what they asked for if they did a ">" - remember that request is always specifying a subject filter.

Reply:
I tend to agree, I'm not sure how much do we really need an additional API for that, considering that you do specify the filter. I would probably leave it as it is.